### ISSUE TITLE
System Error when running Ganache 2.0.0-beta.2 on linux

### ISSUE THREAD

**Issue by zach-is-my-name** [trufflesuite/ganache#1076]
PLATFORM: linux
GANACHE VERSION: 2.0.0-beta.2

EXCEPTION:
```
Error: The module '/home/zmg/Templates/relevant-bonding-curve/node_modules/scrypt/build/Release/scrypt.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 67. This version of Node.js requires
NODE_MODULE_VERSION 57. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:172:20)
    at Object.Module._extensions..node (module.js:598:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:172:20)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/home/zmg/Templates/relevant-bonding-curve/node_modules/scrypt/index.js:3:20)
    at Object.<anonymous> (/home/zmg/Templates/relevant-bonding-curve/node_modules/scrypt/index.js:388:3)
    at Module._compile (module.js:569:30)
    at Module._extensions..js (module.js:580:10)
    at Object.require.extensions.(anonymous function) [as .js] (/home/zmg/Templates/relevant-bonding-curve/node_modules/babel-register/lib/node.js:152:7)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/home/zmg/Templates/relevant-bonding-curve/node_modules/scrypt.js/node.js:1:107)
    at Object.<anonymous> (/home/zmg/Templates/relevant-bonding-curve/node_modules/scrypt.js/node.js:9:3)
    at Module._compile (module.js:569:30)
    at Module._extensions..js (module.js:580:10)
```

**Comment by seesemichaelj** [comment]
@zach-is-my-name Can you do me a favor and tell me what version of node you have (`node --version`)?

I'm guessing this error happened when you tried to load a workspace with your truffle-project in it? (perhaps this project? https://github.com/relevant-community/bonding-curve)

Thanks!

**Comment by zach-is-my-name** [comment]
node v.11.5.0
yes, error occurred then

**Comment by seesemichaelj** [comment]
This is more for dev notes:

So I think the issue is related to the fact that we must `require()` truffle config files to load them since they're not just mere json files. See https://github.com/trufflesuite/ganache/blob/93f9ecab1324f376fa08e42ce0ead4590ebc5ec7/src/truffle-integration/projectDetails.js#L24.

So I think the ganache application, compiled on Travis with node 10, is using that node engine when running that `require()` statement. The truffle config file then requires other modules and node finds `scrypt` in that folder's `node_modules` which was compiled using the user's node (in this case `11.5.0`).

This is a tricky problem...

**Comment by seesemichaelj** [comment]
Thanks for reporting this @zach-is-my-name!

**Comment by seesemichaelj** [comment]
I was able to reproduce this with the following steps:

- Create a truffle project
- `nvm use 11`
- `npm install scrypt`
- Link the project in `ganache@2.0.0-beta.2`

I was curious how `truffle` loads projects which are compiled against different node versions, and then it dawned on me that `truffle` is a globally installed npm module and uses the installed version of `node`. Ganache packages all that up in the distributable and uses it's version of electron (which packages node).

Tricky indeed...going to talk some more with the team on slack

**Comment by seesemichaelj** [comment]
Until we come up with a solution, the only workaround here @zach-is-my-name is for you to use `node@v8` (any 8.x version will work) as your node version for your truffle project. This is the version that the Ganache package was built against

**Comment by zach-is-my-name** [comment]
Thanks for following up. Very interesting diagnostics

**Comment by seesemichaelj** [comment]
This will be fixed in the next release of 2.0 (still not sure if it'll be another beta or not)!
